src/Agent.Worker/Handlers/PowerShell3Handler.cs

```diff
@@ -90,11 +91,17 @@ public async Task RunAsync()
 
         private void OnDataReceived(object sender, ProcessDataReceivedEventArgs e)
         {
+            string line = e.Data ?? string.Empty;
+            if (_shouldRemoveColorsFromLogs)
+            {
+                line = OutputDataHelper.RemoveAnsiColorsFromLine(line);
```

Comment:
I'm thinking it would be better to do this where secrets are scrubbed (e.g. scrub both versions with and w/o the Ansi sequences removed) .  This implementation could potentially cause partial secrets to be missed if the secret itself contains a matching sequence

Also, it would be better if output in the logs was not modified unless it contains a scrubbed secret as removing the escape sequences might cause issues with diagnosability (e.g. if a tool is unexpectedly outputting the sequences which the user needs to know about for debugging)